app/code/Magento/Eav/Model/Attribute/Data/Text.php

```diff
@@ -72,7 +72,7 @@ public function validateValue($value)
             return true;
         }
 
-        if (empty($value) && $value !== '0') {
+        if (empty($value) && $value !== '0' && $attribute->getDefaultValue() == NULL) {
```

Comment:
Why previous logic does not work for `'0'`?

Reply:
@orlangur previously the default value is not checked . So if in case default value is set as '0' then if case is executed.

Reply:
@orlangur ?